circuit/src/builder/circuit_builder.rs

```diff
@@ -700,6 +713,68 @@ mod tests {
         assert_eq!(circuit.primitive_ops.len(), 2);
     }
 
+    #[derive(Debug, Clone)]
+    struct ConstantHint<const C: usize> {
+        inputs: Vec<ExprId>,
+    }
```

Comment:
What do the `inputs` represent here? More generally why would we need a `ConstantHint`? Constants are preprocessed so fine to have them via the `Op::Const` type.